src/main/java/org/truffleruby/core/hash/HashLiteralNode.java

```diff
@@ -89,18 +92,18 @@ public Object execute(VirtualFrame frame) {
 
                 if (stringKeyProfile.profile(RubyGuards.isRubyString(key))) {
                     if (!isFrozen(key)) {
-                        key = freezeNode.call(frame, dupNode.call(frame, key, "dup"), "freeze");
+                        key = callFreeze(frame, callDup(frame, key));
                     }
                 }
 
-                final int hashed = hashNode.hash(frame, key, false);
+                final int hashed = hash(frame, key);
 
                 final Object value = keyValues[n * 2 + 1].execute(frame);
 
                 for (int i = 0; i < n; i++) {
                     if (i < size &&
                             hashed == PackedArrayStrategy.getHashed(store, i) &&
-                            equalNode.callBoolean(frame, key, "eql?", PackedArrayStrategy.getKey(store, i))) {
+                            callEqual(frame, key, PackedArrayStrategy.getKey(store, i))) {
```

Comment:
This should be named `callEql()` or `eql()`, as `equal?` is a different thing in Ruby (it compares identity, not hash keys like `eql?`).